sdk/go/pulumi-language-go/main.go

```diff
@@ -46,36 +53,65 @@ import (
 	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
 )
 
-func findProgram(binary string) (*exec.Cmd, error) {
-	// we default to execution via `go run`
-	// the user can explicitly opt in to using a binary executable by specifying
-	// runtime.options.binary in the Pulumi.yaml
-	if binary != "" {
-		program, err := executable.FindExecutable(binary)
+func hashDir(path string) string {
```

Comment:
`go build` automatically recompiles if dependencies change, including file contents. If we hash the dir, we will spend a lot of time in the `.git` folder, non-content files.

My `pulumi` dir is 8.5GiB and contains 399,072 files. We don't want to hash all of that.